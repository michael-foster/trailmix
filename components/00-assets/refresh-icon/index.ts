export { RefreshIcon } from './refresh-icon';
