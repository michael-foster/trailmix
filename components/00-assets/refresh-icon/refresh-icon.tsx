import React from 'react';
import { SVG } from '@outside-interactive/web-microfrontends.helpers.types';

export const RefreshIcon = ({ classes = '', fillColor = '' }: SVG) => (
  <svg
    className={classes}
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 16 20"
    role="img"
  >
    <path
      d="M8 4V0L3 5l5 5V6a6 6 0 1 1-6 6H0a8 8 0 1 0 8-8Z"
      fill={fillColor}
    />
  </svg>
);
