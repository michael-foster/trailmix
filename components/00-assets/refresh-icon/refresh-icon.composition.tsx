import React from 'react';
import { RefreshIcon } from './refresh-icon';

export const BasicRefreshIcon = () => <RefreshIcon fillColor="white" />;
