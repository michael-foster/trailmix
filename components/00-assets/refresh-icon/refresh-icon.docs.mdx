---
description: 'A RefreshIcon component.'
labels: ['SVG', 'ASSET']
---

import { RefreshIcon } from './refresh-icon';

## React Component for rendering a refresh icon SVG

### Component usage

```js
<RefreshIcon fillColor="#d2a940" />
```

### Using props to customize the icon

Modify the fill color to see it change live:

```js live
<RefreshIcon fillColor="#d2a940" />
```
