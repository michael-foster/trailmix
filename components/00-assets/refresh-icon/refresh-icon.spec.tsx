import React from 'react';
import { render } from '@testing-library/react';
import { BasicRefreshIcon } from './refresh-icon.composition';

describe('BasicRefreshIcon', () => {
  test('snapshot', () => {
    const { asFragment } = render(<BasicRefreshIcon />);

    expect(asFragment()).toMatchSnapshot();
  });
});
