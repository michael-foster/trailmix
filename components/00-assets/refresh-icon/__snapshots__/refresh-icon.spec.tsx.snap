// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicRefreshIcon snapshot 1`] = `
<DocumentFragment>
  <svg
    class=""
    fill="none"
    role="img"
    viewBox="0 0 16 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 4V0L3 5l5 5V6a6 6 0 1 1-6 6H0a8 8 0 1 0 8-8Z"
      fill="white"
    />
  </svg>
</DocumentFragment>
`;
